sprocs/authz_course.sql

```diff
@@ -34,10 +34,10 @@ BEGIN
 
     IF NOT FOUND THEN
         course_role := 'None';
-    END IF;
 
-    IF is_example_course AND allow_example_course_override THEN
-        course_role := 'Viewer';
+        IF is_example_course AND allow_example_course_override THEN
+            course_role := 'Viewer';
+        END IF;
```

Comment:
The previous arrangement made it so that it was difficult to test these changes with the `dev@illinois.edu` user - even if I made that user a course owner, this condition would always trigger and revert me to a viewer. I don't think this change should have any negative impact, but I wanted to call it out for review since it's auth code and that's really important to get right.

Reply:
I checked on prod and we have a lot of users who are `Editor` for XC 101. I'm not sure why this happened, but I'm worried that this change will allow all of them to actually act as editors and not be force-reverted to viewers.

We could delete all course-level permissions for XC 101 in prod, but we should probably check why we have so many Staff listed there. Are we automatically adding users to course staff in XC 101? Or did we used to do this in the past but now we just automatically give people staff access without adding them? I should remember the state of this, but I don't 😭

Reply:
Ah, thank you so much for actually checking!

As far as I can tell, we are not automatically adding users as editors (or anything else) for the example course, though I'm hesitant to say that with certainty. I'd appreciate a second set of eyes on this.

I'll leave this change around for now as I continue to iterate on this PR, but I'm happy to back it out if we can't gain confidence that this would be safe to do.

Reply:
> we are not automatically adding users as editors (or anything else) for the example course

I'm not sure about `Editor`, but we auto-add XC 101 `Viewer` upon attempted access (#2747). I believe the idea was to make it easier to demo questions to non-PL users.

> we have a lot of users who are `Editor` for XC 101. I'm not sure why this happened

Was this due to the PL Workshop a while back?

Reply:
To be clear, I agree with this change. I just want to make sure that it's safe to deploy 😄

I just deleted all XC 101 course staff on prod. Let's wait and see whether any of them re-appear before we merge this PR. If they stay deleted then I think we are safe to deploy this (we'll also need to delete XC 101 staff in other regions).